JMMServer/Providers/MyAnimeList/MALHelper.cs

```diff
@@ -530,15 +532,21 @@ public static void UpdateMALSeries(AnimeSeries ser)
                             ser.GetAnime().AnimeID);
                         continue;
                     }
+
+                    string confirmationMessage = "";
+                    if (animeListHashtable.ContainsKey(malID))
```

Comment:
I don't think there is an existing condition, I am just pointing out that in case of mass update this API call will most likely give you outdated list thus it can fail in case where you delete series but MAL still says it is there, so it will try to update instead of add and will fail.
We can try your code and see if there will be any issues, I am not against it or anything it is just I don't see it being more reliable than existing one.